docs/src/snap/howto/2-node-ha.md

```diff
@@ -0,0 +1,423 @@
+
+See the [external datastore guide] for more information on how Canonical
+Kubernetes can be configured to use other datastores.
+
+## Guide
+
+### Prerequisites
+
+Make sure that:
+
+* Both nodes have joined the Kubernetes cluster.
+  See the [getting started] and [add/remove nodes] guides.
+* The user associated with the HA service has SSH access to the peer node and
+  passwordless sudo configured. For simplicity, the default "ubuntu" user can
+  be used.
+* We recommend using static IP configuration.
+
+The [2ha.sh script] automates most operations related to the 2-node HA scenario.
+Retrieve it like so:
+
+```
+sudo mkdir -p /var/snap/k8s/common
+repo=https://raw.githubusercontent.com/petrutlucian94/k8s-snap
+sudo curl $repo/refs/heads/KU-1606/2ha_script/k8s/hack/2ha.sh \
+    -o /var/snap/k8s/common/2ha.sh
+sudo chmod a+rx /var/snap/k8s/common/2ha.sh
+```
+
+The first step is to install the required packages:
+
+```
+/var/snap/k8s/common/2ha.sh install_packages
+```
+
+### DRBD
+
+For the purpose of this guide, we are going to use a loopback device as DRBD
+backing storage:
+
+```
+sudo dd if=/dev/zero of=/opt/drbd0-backstore bs=1M count=2000
+```
+
+Ensure that the loopback device is attached at boot time, before Pacemaker
+starts.
+
+```
+cat <<EOF | sudo tee /etc/rc.local
+#!/bin/sh
+mknod /dev/lodrbd b 7 200
+losetup /dev/lodrbd /opt/drbd0-backstore
+EOF
+
+sudo chmod +x /etc/rc.local
+
+cat <<EOF | sudo tee /etc/systemd/system/rc-local.service
+# This unit gets pulled automatically into multi-user.target by
+# systemd-rc-local-generator if /etc/rc.local is executable.
+[Unit]
+Description=/etc/rc.local Compatibility
+Documentation=man:systemd-rc-local-generator(8)
+ConditionFileIsExecutable=/etc/rc.local
+After=network.target
+
+[Service]
+Type=forking
+ExecStart=/etc/rc.local start
+TimeoutSec=0
+RemainAfterExit=yes
+GuessMainPID=no
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+sudo systemctl enable rc-local.service
+sudo systemctl start rc-local.service
+```
+
+Let's configure the DRBD block device that will hold the Dqlite data.
+Make sure to use the right node addresses.
+
+```
+# Disable the DRBD service, it will be managed through Pacemaker.
+sudo systemctl disable drbd
+
+HAONE_ADDR=<firstNodeAddress>
+HATWO_ADDR=<secondNodeAddress>
+
+cat <<EOF | sudo tee /etc/drbd.d/r0.res
+resource r0 {
+  on haone {
+    device /dev/drbd0;
+    disk /dev/lodrbd;
+    address ${HAONE_ADDR}:7788;
+    meta-disk internal;
+  }
+  on hatwo {
+    device /dev/drbd0;
+    disk /dev/lodrbd;
+    address ${HATWO_ADDR}:7788;
+    meta-disk internal;
+  }
+}
+EOF
+
+sudo drbdadm create-md r0
+sudo drbdadm status
+```
+
+Let's create a mount point for the DRBD block device. Non-default mount points
+need to be passed to the 2ha.sh script mentioned above, see the script for the
+full list of configurable parameters.
+
+```
+DRBD_MOUNT_DIR=/mnt/drbd0
+sudo mkdir -p $DRBD_MOUNT_DIR
+```
+
+Run the following *once* to initialize the filesystem.
```

Comment:
Maybe add what would happen if you accidentally execute it multiple times?
e.g. "Executing this multiple times would cause ..."

Reply:
Not much since we haven't yet stored any data yet, not sure if it's worth mentioning it.

The main reason why I emphasised "once" is that below there's a line saying ``Follow the above steps on both nodes before moving forward.``, however this step in particular doesn't have to be executed twice.